gprofiler/py_package_version.py

```diff
@@ -0,0 +1,160 @@
+"""Package name and version for Python files.
+
+Some of the functions in this module are implemented based on similar functions
+in pip: _get_metadata, _convert_legacy_entry, _files_from_record,
+_files_from_legacy, _get_package_name.
+"""
+import csv
+import email
+import os
+import pathlib
+from typing import Iterator, List, Optional, Tuple
+
+import pkg_resources
+
+# A little monkey patch to prevent pkg_resources from converting "/proc/{pid}/root/" to "/"
+pkg_resources._normalize_cached = lambda path: path
+
+
+__all__ = ["get_versions"]
+
+
+def _convert_to_proc_root_path(path: str, pid: int) -> str:
+    assert path.startswith("/")
+    return os.path.join(f"/proc/{pid}/root", path)
+
+
+def _get_packages_dir(file_path: str) -> Optional[str]:
+    if not file_path.startswith("/"):
+        return None
+
+    idx = file_path.rfind("-packages/")
```

Comment:
Maybe just `rfind` for `site-packages`/`dist-packages`, it will be a lot more simple and readable

Reply:
Used `rpartition` instead